impeller/toolkit/glvk/trampoline.cc

```diff
@@ -0,0 +1,328 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "impeller/toolkit/glvk/trampoline.h"
+
+#include <array>
+
+#include "flutter/fml/closure.h"
+#include "flutter/fml/trace_event.h"
+#include "impeller/base/timing.h"
+#include "impeller/base/validation.h"
+#include "impeller/geometry/point.h"
+#include "impeller/renderer/backend/gles/formats_gles.h"
+#include "impeller/toolkit/android/proc_table.h"
+#include "impeller/toolkit/egl/image.h"
+
+namespace impeller::glvk {
+
+static GLuint kAttributeIndexPosition = 0u;
+static GLuint kAttributeIndexTexCoord = 1u;
+
+static constexpr const char* kVertShader = R"IMPELLER_SHADER(#version 100
+
+precision mediump float;
+
+attribute vec2 aPosition;
+attribute vec2 aTexCoord;
+
+varying vec2 vTexCoord;
+
+void main() {
+  gl_Position = vec4(aPosition, 0.0, 1.0);
+  vTexCoord = aTexCoord;
+
+}
+)IMPELLER_SHADER";
+
+static constexpr const char* kFragShader = R"IMPELLER_SHADER(#version 100
+
+#extension GL_OES_EGL_image_external : require
+
+precision mediump float;
+
+uniform samplerExternalOES uTexture;
```

Comment:
My understanding was that surface textures would never give us non external textures. Is that not the case.

Reply:
Oh, you meant in terms of the trampoline being a general purpose utility. Thats a good point, I can patch it to work with both. It will probably require another shader but thats fine.

Reply:
Its possible to do something like create a SurfaceTexture, grab the surface from that, lock canvas and draw with the Android canvas, then give that SurfaceTexture to flutter. That would not be an oes texture.

Reply:
TBH, I was just following what we did with Skia in GL and that is GL only. https://github.com/flutter/engine/blob/207f088f73cb00fa05aff170030df891d62cadf5/shell/platform/android/surface_texture_external_texture_gl_skia.cc#L43

Reply:
Its not OES only though, whatever that configuration happens to say. Perhaps the oes texture binding is just very permissive?